static/js/settings_account.js

```diff
@@ -129,6 +129,11 @@ exports.intialize_custom_user_type_fields = function (element_id, user_id, is_ed
     var field_types = page_params.custom_profile_field_types;
     var user_pills = {};
 
+    var person = people.get_person_from_user_id(user_id);
+    if (person.is_bot) {
+        return null;
+    }
```

Comment:
Can you explain why additional `if` is required here? This function is calling `get_custom_profile_data`, which returns null if user is bot. 
I think it must be raising error at?
```
if (field_value_raw) {
       field_value_raw = field_value_raw.value;
}
```

Reply:
> Can you explain why additional if is required here?

The error is because of lines 147-148:
```javascript
var pill_container = $(element_id).find('.custom_user_field[data-field-id="' + field.id + '"].pill-container').expectOne();
```

Namely the .expectOne() part since for bots, the pill container was not being created in the first place.

> I think it must be raising error at?

Can't be the case since as you've said, `people.get_custom_profile_data()` will be returning null if the user is a bot (on line 138). In which case `field_value_raw` would just be null and then that line `field_value_raw = field_value_raw.value;` would just be skipped. Right? Or have I misinterpreted your question?

Reply:
No, get it. I was thinking that if `field_value_raw` is null it shouldn't execute `if..` condition to render. But `is_editable` flag is true, so that is the reason it satisfies that condition.

Reply:
Shouldn't we return an empty list, rather than `null`?  IT seems like that's what the only caller that looks at the return type does.

Also can you add another commit fixing the typo in this name: `intialize_custom_user_type_fields` (should be "initialize")?

Reply:
> Shouldn't we return an empty list, rather than null?
Yup, that makes works too. Plus it makes more sense, so I've made the change :)